core/indigo-core/molecule/src/base_molecule.cpp

```diff
@@ -4325,3 +4313,81 @@ void BaseMolecule::getBoundingBox(Rect2f& bbox) const
     }
     bbox = Rect2f(a, b);
 }
+
+void MetaDataStorage::addMetaObject(MetaObject* pobj)
+{
+    int index = _meta_data.size();
+    _meta_data.expand(index + 1);
+    _meta_data.set(index, pobj);
+
+    switch (pobj->_class_id)
+    {
+    case KETSimpleObject::CID:
+        _simple_object_indexes.push() = index;
+        break;
+    case KETReactionPlus::CID:
+        _plus_indexes.push() = index;
+        break;
+    case KETReactionArrow::CID:
+        _arrow_indexes.push() = index;
+        break;
+    default:
+        break;
+    }
+}
+
+void MetaDataStorage::cloneMetaData(const MetaDataStorage& other)
+{
+    resetMetaData();
+    const auto& meta = other.metaData();
+    for (int i = 0; i < meta.size(); i++)
+        addMetaObject(meta[i]->clone());
+}
+
+const MetaObject& MetaDataStorage::getMetaObject(std::uint32_t meta_type, int index) const
+{
+    switch (meta_type)
+    {
+    case KETSimpleObject::CID:
+        return *_meta_data[_simple_object_indexes[index]];
+        break;
+    case KETReactionPlus::CID:
+        return *_meta_data[_plus_indexes[index]];
+        break;
+    case KETReactionArrow::CID:
+        return *_meta_data[_arrow_indexes[index]];
+        break;
+    default:
+        break;
```

Comment:
Should return some object's reference or throw an exception.

Reply:
Fixed